consensus/src/consensus_observer/network_message.rs

```diff
@@ -166,8 +163,139 @@ impl ConsensusObserverDirectSend {
 /// OrderedBlock message contains the ordered blocks and the proof of the ordering
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
 pub struct OrderedBlock {
-    pub blocks: Vec<Arc<PipelinedBlock>>,
-    pub ordered_proof: LedgerInfoWithSignatures,
+    blocks: Vec<Arc<PipelinedBlock>>,
+    ordered_proof: LedgerInfoWithSignatures,
+}
+
+impl OrderedBlock {
+    pub fn new(blocks: Vec<Arc<PipelinedBlock>>, ordered_proof: LedgerInfoWithSignatures) -> Self {
+        Self {
+            blocks,
+            ordered_proof,
+        }
+    }
+
+    /// Returns a reference to the ordered blocks
+    pub fn blocks(&self) -> &Vec<Arc<PipelinedBlock>> {
+        &self.blocks
+    }
+
+    /// Returns a copy of the first ordered block
+    pub fn first_block(&self) -> Option<Arc<PipelinedBlock>> {
+        self.blocks.first().cloned()
+    }
+
+    /// Returns a copy of the last ordered block
+    pub fn last_block(&self) -> Option<Arc<PipelinedBlock>> {
+        self.blocks.last().cloned()
+    }
```

Comment:
The vec is not expected to be empty right? wondering if we can unwrap here instead.

Reply:
Yeah, we can do that, too. Updated 😄